jsonnet/kube-thanos/kube-thanos-rule.libsonnet

```diff
@@ -183,4 +183,41 @@ local k = import 'ksonnet/ksonnet.beta.4/k.libsonnet';
       },
     },
   },
+
+  withAlertmanagers:: {
+    local tr = self,
+    config+:: {
+      rulesConfigMapName: error 'must provide rulesConfigMapName',
+      alertmanagersURL: error 'must provide alertmanagersURL',
+    },
+
+    statefulSet+: {
+      spec+: {
+        template+: {
+          spec+: {
+            containers: [
+              if c.name == 'thanos-rule' then c {
+                args+: [
+                  '--alertmanagers.url=' + alertmanagerURL,
+                  for alertmanagerURL in tr.config.alertmanagersURL
+                ] + [
+                  '--rule-file=/etc/thanos/rules/*.rules.yaml',
```

Comment:
This assumes that the key used in the configmap ends in `rules.yaml`, which is mysterious because it is not communicated to the user. Can we instead use the pattern of requiring a `rulesFileKey` in the jsonnet config and use that name instead?

Reply:
Great comments. I would like to just use `*.yaml` instead. Is that OK?

Reply:
The trouble is that there is no guarantee that the configmap key will even end in `.yaml`. The user could write a configmap where the keys are
```yaml
foo: ...
bar: ...
rules: ...
```
each of these keys could have a valid rules file for the value, but the keys don't end in `.yaml` and thus this flag wouldn't work. File extensions are helpful hints and conventions but are not required.
Similarly, what if the configmap hold multiple keys that end in `.yaml` but only ONE of them is actually a rules file? Then the ruler would be attempting to read something it can't understand, potentially causing problems.